### PR TITLE
Equipment loading optimization from db

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 1]
```diff
@@ -306,26 +290,61 @@ public HvdcShuntCompensatorsInfos getHvcLineShuntCompensators(UUID networkUuid,
 
     public List<String> getElementsIds(UUID networkUuid, String variantId, List<String> substationsIds, ElementType elementType) {
         switch (elementType) {
+            case SUBSTATION:
+                return getSubstationsIds(networkUuid, variantId);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:294]: Filtering with substationsIds parameter if this parameter is not null ?

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 2]
```diff
@@ -306,26 +290,61 @@ public HvdcShuntCompensatorsInfos getHvcLineShuntCompensators(UUID networkUuid,
...
+                .flatMap(substationId -> network.getSubstation(substationId).getVoltageLevelStream())
+                .flatMap(voltageLevel -> voltageLevel.getConnectableStream((Class<? extends Connectable>) elementType.getElementClass()))
+                .map(Connectable::getId)
+                .distinct()
+                .collect(Collectors.toList());
+        }
+    }
+
+    public Stream<? extends Connectable> getConnectableStream(Network network, ElementType elementType) {
+        switch (elementType) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:321]: why not using network.getConnectableStream(elementType.getElementClass()) ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:294]: It was already like that

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/network/map/NetworkMapService.java:321]: ah ok

**Pull request merged**